ui/raidboss/emulator/data/PopupTextAnalysis.js

```diff
@@ -1,20 +1,68 @@
 import EmulatorCommon from '../EmulatorCommon';
 import StubbedPopupText from '../overrides/StubbedPopupText';
 
-export default class PopupTextAnalysis extends StubbedPopupText {
-  OnTriggerInternal(trigger, matches, currentTime) {
-    this.currentTriggerStatus = {
-      initialData: EmulatorCommon.cloneData(this.data),
-      condition: undefined,
-      response: false,
-      result: false,
-      delay: false,
-      suppressed: false,
-      executed: false,
-      promise: undefined,
+class Resolver {
+  constructor() {
+    this.promise = null;
+    this.run = null;
+    this.delayTo = null;
+  }
+  async IsResolved(log) {
+    if (this.delayTo) {
+      if (this.delayTo < log.timestamp) {
+        this.delayTo = null;
+        this.delay();
+        await this.delayPromise;
+      } else {
+        return false;
+      }
+    }
+    if (!this.delayTo) {
```

Comment:
this.delayTo is always false here, right? I'm not sure I understand this conditional or the return false at the bottom of this function.

Reply:
Yes, `this.delayTo` is always false-y here. This is a leftover from a previous iteration, I'll clean it up.